app/grandchallenge/workstations/static/workstations/js/session-control.mjs

```diff
@@ -0,0 +1,132 @@
+function openWorkstationSession(element) {
+    return (event) => {
+        const windowIdentifier = element.dataset.workstationWindowIdentifier;
```

Comment:
You might want to disable the button when this function is running so that it cannot be triggered multiple times by impatient users. Removing the disabled attribute should then also be done in the catch block as with the spinner removal.